qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,591 @@
+    elif q == r:
+        b_s = _edge_operator_bi(edge_list, s)
+        a_ps = _edge_operator_aij(edge_list, p, s)
+        a_ps = -a_ps if s < p else a_ps
+        qubit_op = (a_ps * b_s + b_p * a_ps) * (id_op - b_q)
+        final_coeff = 1j * -0.25
+    elif q == s:
+        b_r = _edge_operator_bi(edge_list, r)
+        a_pr = _edge_operator_aij(edge_list, p, r)
+        a_pr = -a_pr if r < p else a_pr
+        qubit_op = (a_pr * b_r + b_p * a_pr) * (id_op - b_q)
+        final_coeff = 1j * 0.25
+    else:
+        raise ValueError(f"unexpected sequence of indices: {p}, {q}, {r}, {s}")
+    return (final_coeff * h2_pqrs) * qubit_op
+
+
+def _unpack_term(
+    term_str: str, expand_number_op: bool = False
+) -> Tuple[Tuple[int, int, int], List[Tuple[int, str]]]:
+    """Return a tuple specifying the counts of kinds of operators in `term_str` and
+    a list of the factors and their indices in `term_str`.
+
+    The factors are represented by tuples of the form `(i, c)`, where `i` is an index
+    and `c` is a character.
+    Allowed characters in `term_str` are 'N+-I`.
+    The returned tuple contains counts for `N`, `+`, and `-`, in that order. Identity operators
+    are ignored.
+
+    Args:
+       `term_str`: a string of characters in `N+-I`.
+       `expand_number_op`: if `True`, number operators are expanded to `(i, '+')`, `(i, '-')`
+         in the returned list of factors.
+
+    Returns:
+      tuple: A tuple of two elements. First, a tuple of three integers giving the number of
+      number, raising, and lowering operators. Second a list of factors represented by
+      tuples of two elements: the first is an index and the second one of "-", "+", or "N".
+      If `expand_number_op` is `True`, then factors of `N` are expanded.
+    """  # pylint: disable=missing-raises-doc
+    (n_number, n_raise, n_lower) = (0, 0, 0)
+    facs = []
+    for i, c in enumerate(term_str):
+        if c == "I":
+            continue
+        if c == "+":
+            n_raise += 1
+            facs.append((i, "+"))
+        elif c == "-":
+            n_lower += 1
+            facs.append((i, "-"))
+        elif c == "N":
+            n_number += 1
+            if expand_number_op:
+                facs.append((i, "+"))
+                facs.append((i, "-"))
+            else:
+                facs.append((i, "N"))
+        else:
+            raise ValueError("Unexpected operator ", c, " in term.")
+
+    return (n_number, n_raise, n_lower), facs
+
+
+def _interaction_type(n_number: int, n_raise: int, n_lower: int) -> TermType:
+    """Return a `TermType` instance describing the type of interaction given the number of
+    number, raising, and lowering operators.
+
+    The number number operators must be 1 or 2. The number of raising operators
+    must be equal to 0, 1 or 2, and the number of raising operators must be equal
+    to the number of lowering operators.
+
+    Args:
+       `n_number`: the number of number operators
```

Comment:
And here - but since we don't emit docs for private methods presently whether this would change the outcome in the html or not I am uncertain.

Reply:
Still a good idea to follow the convention.